File: 3152. Special Array II/solution.py
class Solution:
    def parity(self,x:int):
        if x%2 == 0:
            return True
        return False   
    def isArraySpecial(self, nums: List[int], queries: List[List[int]]) -> List[bool]:
        count = 0
        prefix_array = [0]
        for i  in range(1, len(nums)):
            if self.parity(nums[i-1]) == self.parity(nums[i]):
                count += 1
            prefix_array.append(count)
        res = []
        for start , end in queries:
            if prefix_array[end] -  prefix_array[start]  == 0:
                res.append(True)
            else:
                res.append(False)    
        return res


            

        

File: 3152. Special Array II/solution.rs
impl Solution {
    pub fn parity(i : i32) -> bool{
        if i%2 == 0{
            return true
        }
        false
    }
    pub fn is_array_special(nums: Vec<i32>, queries: Vec<Vec<i32>>) -> Vec<bool> {
        let mut prefix_sum = vec![0];
        let mut count = 0;
        for i in 1..nums.len(){
            if Self::parity(nums[i]) == Self::parity(nums[i-1]){
                count += 1;
            }
            prefix_sum.push(count);
        }
        let mut res = vec![];
        for num in queries{
            let start:usize = num[0] as usize;
            let end:usize = num[1] as usize;
            res.push(prefix_sum[end]-prefix_sum[start]==0);
        } 
        res
    }
}